pygmt/helpers/decorators.py

```diff
@@ -757,15 +757,12 @@ def new_module(*args, **kwargs):
                 else:
                     continue
 
-                issequence = fmt in separators
-                if issequence and is_nonstr_iter(value):
+                if fmt in separators and is_nonstr_iter(value):
                     for index, item in enumerate(value):
-                        try:
+                        if " " in str(item):
                             # Check if there is a space " " when converting
                             # a pandas.Timestamp/xr.DataArray to a string.
                             # If so, use np.datetime_as_string instead.
-                            assert " " not in str(item)
-                        except AssertionError:
```

Comment:
Yeah, looking back at https://github.com/GenericMappingTools/pygmt/pull/562#discussion_r470963848, not sure why I went with try-except instead of just an if-then check :laughing: